kore-rpc-types/src/Kore/JsonRpc/Error.hs

```diff
@@ -0,0 +1,47 @@
+module Kore.JsonRpc.Error (module Kore.JsonRpc.Error) where
+
+import Control.Exception (ErrorCall (..), SomeException)
+import Control.Monad.Logger (logInfoN)
+import Data.Aeson
+import Data.Text qualified as Text
+import Kore.JsonRpc.Server (ErrorObj (..), JsonRpcHandler (..))
+import Text.Casing (fromHumps, toWords)
+
+-- RPC Server implementation errors
+
+cancelUnsupportedInBatchMode, notImplemented :: ErrorObj
+cancelUnsupportedInBatchMode = ErrorObj "Cancel request unsupported in batch mode" (-32001) Null
+notImplemented = ErrorObj "Not implemented" (-32002) Null
+
+unsupportedField :: Value -> ErrorObj
+unsupportedField = ErrorObj "Unsupported option" (-32003)
+
+-- Runtime backend errors
+
+data JsonRpcBackendError
+    = RuntimeError
+    | CouldNotParsePattern
+    | CouldNotVerifyPattern
+    | CouldNotFindModule
+    | ImplicationCheckError
+    | SmtSolverError
+    | Aborted
+    | MultipleStates
+    deriving stock (Enum, Show)
+
+backendError :: ToJSON a => JsonRpcBackendError -> a -> ErrorObj
```

Comment:
I'd suggest to not use `0` as an error code (just by convention), by adding a `NotAnError` constructor in the beginning. 
Also, we should have a comment on this datatype to only ever add errors at the end so we don' t accidentally change existing error codes (since we use `fromEnum`).